src/verification/mod.rs

```diff
@@ -118,6 +118,7 @@ pub(crate) struct JwtClaimsVerifier<'a, K>
 where
     K: JsonWebKey,
 {
+    allowed_jose_types: Option<Vec<JsonWebTokenType>>,
```

Comment:
based on my comment in #160, let's just change this to `is_typ_check_enabled: bool`